R/juvenile_month_dynamic.R

```diff
@@ -0,0 +1,419 @@
+                     dimnames = list(lateFallRunDSM::watershed_labels, 
+                                     lateFallRunDSM::size_class_labels))
+  
+  if (month == 11) {
+    # all remaining fish outmigrate
+    sutter_fish <- migrate(sutter_fish, migratory_survival$sutter, stochastic = stochastic)
+    upper_mid_sac_fish <- migrate(upper_mid_sac_fish + juveniles[1:15, ], migratory_survival$uppermid_sac, stochastic = stochastic)
+    migrants[1:15, ] <- upper_mid_sac_fish + sutter_fish
+    yolo_fish <- migrate(yolo_fish, migratory_survival$yolo, stochastic = stochastic)
+    migrants[18:20, ] <- juveniles[18:20, ] + yolo_fish
+    lower_mid_sac_fish <- migrate(lower_mid_sac_fish + migrants, migratory_survival$lowermid_sac, stochastic = stochastic)
+    migrants <- lower_mid_sac_fish
+    migrants[23, ] <- juveniles[23, ]
+    lower_sac_fish <- migrate(lower_sac_fish + migrants, migratory_survival$lower_sac, stochastic = stochastic)
+    migrants[25:27, ] <- juveniles[25:27, ]
+    san_joaquin_fish <- migrate(juveniles[28:30, ] + san_joaquin_fish, migratory_survival$san_joaquin, stochastic = stochastic)
+    migrants[18:20, ] <- migrants[18:20, ] + yolo_fish
+    migrants[28:30, ] <- san_joaquin_fish
+    
+    delta_fish <- route_and_rear_deltas(year = year, month = month,
+                                        migrants = round(migrants),
+                                        north_delta_fish = north_delta_fish,
+                                        south_delta_fish = south_delta_fish,
+                                        north_delta_habitat = habitat$north_delta,
+                                        south_delta_habitat = habitat$south_delta,
+                                        freeport_flows = ..params$freeport_flows,
+                                        cc_gates_days_closed = ..params$cc_gates_days_closed,
+                                        rearing_survival_delta = rearing_survival$delta,
+                                        migratory_survival_delta = migratory_survival$delta,
+                                        migratory_survival_sac_delta = migratory_survival$sac_delta,
+                                        migratory_survival_bay_delta = migratory_survival$bay_delta,
+                                        juveniles_at_chipps = juveniles_at_chipps,
+                                        growth_rates = lateFallRunDSM::params$growth_rates,
+                                        territory_size = lateFallRunDSM::params$territory_size,
+                                        stochastic = stochastic)
+    
+    
+    migrants_at_golden_gate <- delta_fish$migrants_at_golden_gate
+    
+  } else {
+    
+    if (hypothesis == 2 & month == 4) {
+      # send specified proportion of fry out of tributaries
+      leave_battle <- if (stochastic) {
+        rbinom(1, juveniles[3, ], ..params$prob_fry_leave)
+      } else {
+        juveniles[3, ] * ..params$prob_fry_leave
```

Comment:
should round to be consistent with other deterministic code...although it shouldn't matter much